src/ImageSharp/Common/Helpers/SimdUtils.Avx2Intrinsics.cs

```diff
@@ -0,0 +1,103 @@
+                            source.Slice(0, adjustedCount),
+                            dest.Slice(0, adjustedCount));
+
+                        source = source.Slice(adjustedCount);
+                        dest = dest.Slice(adjustedCount);
+                    }
+                }
+            }
+
+            /// <summary>
+            /// Implementation of <see cref="SimdUtils.BulkConvertNormalizedFloatToByteClampOverflows"/>, which is faster on new .NET runtime.
+            /// </summary>
+            /// <remarks>
+            /// Implementation is based on MagicScaler code:
+            /// https://github.com/saucecontrol/PhotoSauce/blob/a9bd6e5162d2160419f0cf743fd4f536c079170b/src/MagicScaler/Magic/Processors/ConvertersFloat.cs#L453-L477
+            /// </remarks>
+            internal static void BulkConvertNormalizedFloatToByteClampOverflows(
+                ReadOnlySpan<float> source,
+                Span<byte> dest)
+            {
+                VerifySpanInput(source, dest, Vector256<byte>.Count);
+
+                int n = dest.Length / Vector256<byte>.Count;
+
+                ref Vector256<float> sourceBase =
+                    ref Unsafe.As<float, Vector256<float>>(ref MemoryMarshal.GetReference(source));
+                ref Vector256<byte> destBase = ref Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(dest));
+
+                var maxBytes = Vector256.Create(255f);
+                ref byte maskBase = ref MemoryMarshal.GetReference(PermuteMaskDeinterleave8x32);
+                Vector256<int> mask = Unsafe.As<byte, Vector256<int>>(ref maskBase);
+
+                for (int i = 0; i < n; i++)
+                {
+                    ref Vector256<float> s = ref Unsafe.Add(ref sourceBase, i * 4);
+
+                    Vector256<float> f0 = s;
+                    Vector256<float> f1 = Unsafe.Add(ref s, 1);
+                    Vector256<float> f2 = Unsafe.Add(ref s, 2);
+                    Vector256<float> f3 = Unsafe.Add(ref s, 3);
+
+                    Vector256<int> w0 = ConvertToInt32(f0, maxBytes);
+                    Vector256<int> w1 = ConvertToInt32(f1, maxBytes);
+                    Vector256<int> w2 = ConvertToInt32(f2, maxBytes);
+                    Vector256<int> w3 = ConvertToInt32(f3, maxBytes);
+
+                    Vector256<short> u0 = Avx2.PackSignedSaturate(w0, w1);
+                    Vector256<short> u1 = Avx2.PackSignedSaturate(w2, w3);
+                    Vector256<byte> b = Avx2.PackUnsignedSaturate(u0, u1);
+                    b = Avx2.PermuteVar8x32(b.AsInt32(), mask).AsByte();
+
+                    Unsafe.Add(ref destBase, i) = b;
+                }
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            private static Vector256<int> ConvertToInt32(Vector256<float> vf, Vector256<float> scale)
+            {
+                vf = Avx.Multiply(vf, scale);
+                return Avx.ConvertToVector256Int32(vf);
+            }
```

Comment:
A couple of things I noticed. You're using `Avx` over `Avx2`. I've noticed that in @saucecontrol -s  original source also. Wouldn't `Avx2` be faster? 

We're not clamping like we do in the other implementations. I had a quick go. (I don't know how `Vector256<float>.Zero` is treated here, should it be passed as a param?) but there's still minor differences. 

```c#
private static Vector256<int> ConvertToInt32(Vector256<float> vf, Vector256<float> scale, Vector256<float> offset)
{
    vf = Avx2.Multiply(vf, scale);
    vf = Avx2.Add(vf, offset);
    vf = Avx2.Min(Avx2.Max(vf, Vector256<float>.Zero), scale);
    return Avx2.ConvertToVector256Int32(vf);
}
```

Reply:
`Avx2` inherits from `Avx`, and that `Multiply` overload is not part of the AVX2 instruction set, you're just accessing it through the `Avx2` class here, but you're calling the same method. In fact if you use Re# it does suggest to simplify the name and just use `Avx` 👍

Reply:
@JimBobSquarePants new x86 instructions sets are extensions to previous families of instructions. Despite the fact that these methods are static, designers of `System.Runtime.Intrinsics` have chosen to use nonstatic classes with a [nice inheritance chain](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.x86.avx2?view=netcore-3.1) to model this.  `Multiply(Vector256<float>, Vector256<float>)` is in fact defined in the base class `Avx`, because the backing `VMULPS ymm, ymm, ymm/m256` instruction is already available in AVX. New x86 CPU families improve the performance of existing instructions instead of replacing them with new ones.

Reply:
I wish the docs were better. You have to look up each method on the intel docs which is just misdirection.

Reply:
I don't think it makes sense to duplicate the Intel content because the specification is owned and maintained by Intel. Some instructions do crazy complex stuff. (See: specification of [`Avx2.PackUnsignedSaturate` / `_mm256_packus_epi16`](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm256_packus_epi16&expand=4118) under "Operation"). Other instructions are trivially simple (`Add`, `Multiply`), while still best specified by the original docs.

Adding a link to related Intel docs would be nice though.

Reply:
> Wouldn't Avx2 be faster

In general AVX instructions operate on floats and AVX2 operate on integer types.  If seeing them mixed in code bothers you, you can simply `using static System.Runtime.Intrinsics.X86.Avx2` and have access to everything without the class name.  I personally find it useful to specify the exact ISA for each instruction because it makes it easier to spot when an `IsSupported` check is missing.

Reply:
I just realised VS actually suggests `Avx` anyway, offering to simplify if you use `Avx2`. Pretty neat!